### PR TITLE
Support hapi 17

### PR DESCRIPTION
**Pull request by GunnarHolwerda**
- Adds support for hapi 17 options key in route configuration
- Upgrades all dependencies to latest versions to resolve security issues (sorry docpress 😞)
- Adds typings

I spent a lot of time trying to decide how to support Hapi < 17 and decided with defaulting the `optionsKey` to `'config'` if you didn't want a breaking version. I created a new decorator `cntrlr` that defaults to the `'options'` key. If you can think of a better way of providing this configuration I would be really happy!

I understand I made a lot of sweeping changes here, so I totally understand if you would like me to restructure my PR. Just let me know!

At this point I also don't know which versions of node you would want to support. I added some of the later versions, but unfortunately some of the new updates have broken the older versions.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,11 +5,11 @@ node_js:
   - '0.12'
   - 'iojs-1'
   - 'iojs-2'
-  - '5',
-  - '6',
-  - '7',
-  - '8',
-  - '9',
+  - '5'
```
Comment (knownasilya) [inline .travis.yml]: I think we want to do node 4 and up, since that is the supported version by node.

File: .travis.yml [hunk 2]
```diff
@@ -5,5 +5,11 @@ node_js:
   - '0.12'
   - 'iojs-1'
   - 'iojs-2'
+  - '5'
```
Comment (knownasilya) [inline .travis.yml]: I think 4+ is probably the best way to go since that's the minimum supported.

File: README.md [hunk 3]
```diff
@@ -69,11 +69,15 @@ use the [transform-decorators-legacy] plugin. See this [post] for detailed instr
 
 ## Decorators
 
-### `@controller(basePath)`
+### `@controller(basePath, optionsKey = 'config')`
```
Comment (knownasilya) [inline README.md]: I don't think we need to do this, I'm for doing a major change and bumping it to v1 (we could do v0.5, but v1 would be more curious to existing users). So defaulting to what v17 Hapi support and up for v1.

File: index.js [hunk 4]
```diff
@@ -39,14 +40,19 @@ exports.controller = function controller (baseUrl) {
...
+        hapiRoute[optionsKey] = hapiRoute.settings
+        hapiRoute[optionsKey].bind = self
 
         return hapiRoute
       })
     }
   }
 }
 
+exports.cntrlr = function (baseUrl) {
```
Comment (knownasilya) [inline index.js]: We should be able to remove this as well.

File: package.json [hunk 5]
```diff
@@ -27,21 +28,24 @@
...
     "extend": "^3.0.0",
-    "in-publish": "^2.0.0",
-    "lodash.find": "^3.2.1"
+    "in-publish": "^2.0.0"
   },
   "devDependencies": {
-    "babel": "^5.8.23",
-    "babel-eslint": "^5.0.4",
-    "coveralls": "^2.11.4",
-    "docpress": "^0.6.13",
```
Comment (knownasilya) [inline package.json]: We can keep this for now, since it's devDep and only used for generating the site. I'll make the security changes on that end.

**Comment by GunnarHolwerda** [comment]
Thanks for the feedback! I will get right on fixing this up!

**Comment by GunnarHolwerda** [comment]
@knownasilya Hopefully this all looks okay now!

File: .travis.yml [hunk 6]
```diff
@@ -1,9 +1,11 @@
 language: node_js
 node_js:
-  - '0.10'
-  - '0.11'
-  - '0.12'
-  - 'iojs-1'
-  - 'iojs-2'
+  - '4'
+  - '5'
+  - '6'
```
Comment (knownasilya) [inline .travis.yml]: On last change, can you remove v4 and v5, those are EOL by node so I'd like to remove them from this major release as well.

**Comment by knownasilya** [comment]
I let this fall through the gaps, but can merge and release today if you can make that last tweak. 👍

**Comment by GunnarHolwerda** [comment]
Awesome! I finished making that last tweak!

File: .babelrc [hunk 7]
```diff
@@ -1,4 +1,5 @@
 {
+  "presets": ["es2015"],
```
Comment (knownasilya) [inline .babelrc]: Do we still need this now that we're using node 6+?

Comment (knownasilya) [reply .babelrc]: Probably the last thing haha

**Comment by GunnarHolwerda** [comment]
Hahaha should be all good now! Thanks :smile:

**Comment by knownasilya** [comment]
v1.0.0 has been published

**Pull request merged**